test/integration/aws_lambda_filter_integration_test.cc

```diff
@@ -39,14 +71,71 @@ class AWSLambdaFilterIntegrationTest
    * Initializer for an individual integration test.
    */
   void initialize() override {
-    if (use_chain_) {
+
+    if (add_transform_) {
+      // not sure why but checking the "authorization" in the test succeeds.
+      // what i really want to test is that the filter that follows the aws filter gets the auth header
+      // so i use transformation to copy the auth header to another header, and test that header instead.
+      config_helper_.prependFilter(R"EOF(
+name: io.solo.aws_lambda
+typed_config:
+  "@type": type.googleapis.com/envoy.api.v2.filter.http.FilterTransformations
+  transformations:
+  - match:
+      prefix: /
+    route_transformations:
+      request_transformation:
+        transformation_template:
+          advanced_templates: false
+          headers:
+            x-authorization:
+              text: '{{header("authorization")}}'
+          passthrough: {}
+)EOF");
+
+    }
+    switch (cred_mode_) {
+      case CredMode::CHAIN:
       // set env vars for test
       TestEnvironment::setEnvVar("AWS_ACCESS_KEY_ID", "access key", 1);
       TestEnvironment::setEnvVar("AWS_SECRET_ACCESS_KEY", "access key", 1);
       TestEnvironment::setEnvVar("AWS_SESSION_TOKEN", "session token", 1);
-      config_helper_.addFilter(USE_CHAIN_LAMBDA_FILTER);
-    } else {
-      config_helper_.addFilter(DEFAULT_LAMBDA_FILTER);
+      config_helper_.prependFilter(USE_CHAIN_LAMBDA_FILTER);
+      break;
+    case CredMode::STS:
+      // set env vars for test
+      TestEnvironment::setEnvVar("AWS_WEB_IDENTITY_TOKEN_FILE", TestEnvironment::runfilesPath("test/integration/fakejwt.txt", "envoy_gloo"), 1);
+      TestEnvironment::setEnvVar("AWS_ROLE_ARN", "test", 1);
+      config_helper_.prependFilter(USE_STS_LAMBDA_FILTER);
+      break;
+    case CredMode::DEFAULT:
+      config_helper_.prependFilter(DEFAULT_LAMBDA_FILTER);
+    }
+
+    if (add_transform_) {
```

Comment:
why have two separate checks here for `if (add_transform_) {` and then prepend filter? seems like these could live in a single if block